pkg/app/piped/controller/scheduler.go

```diff
@@ -537,6 +538,23 @@ func (s *scheduler) executeStage(sig executor.StopSignal, ps model.PipelineStage
 		return model.StageStatus_STAGE_FAILURE
 	}
 
+	// Skip the stage if needed based on the skip config.
+	skip, err := s.checkSkipStage(sig.Context(), input)
```

Comment:
If we don't need the executor to check whether this stage is skippable, we should run it before L532 to reduce the waste.